rmgpy/reaction.py

```diff
@@ -555,27 +555,50 @@ def get_free_energy_of_reaction(self, T):
                 raise
         return dGrxn
 
-    def get_equilibrium_constant(self, T, type='Kc'):
+    def get_equilibrium_constant(self, T, type='Kc', surface_site_density=2.5e-05):
         """
         Return the equilibrium constant for the reaction at the specified
-        temperature `T` in K. The `type` parameter lets	you specify the
-        quantities used in the equilibrium constant: ``Ka`` for	activities,
-        ``Kc`` for concentrations (default), or ``Kp`` for pressures. Note that
-        this function currently assumes an ideal gas mixture.
+        temperature `T` in K and reference `surface_site_density` 
+        in mol/m^2 (2.5e-05 default) The `type` parameter lets you specify 
+        the quantities used in the equilibrium constant: ``Ka`` for activities,
+        ``Kc`` for concentrations (default), or ``Kp`` for pressures.  This
+        function assumes a reference pressure of 1e5 Pa for gas phases species
+        and uses the ideal gas law to determine reference concentrations. For 
+        surface site species, a reference concentration of 1 is assumed. 
         """
         cython.declare(dGrxn=cython.double, K=cython.double, C0=cython.double, P0=cython.double)
         # Use free energy of reaction to calculate Ka
         dGrxn = self.get_free_energy_of_reaction(T)
         K = np.exp(-dGrxn / constants.R / T)
         # Convert Ka to Kc or Kp if specified
+        # Assume a pressure of 1e5 Pa for gas phase species
         P0 = 1e5
+        # Determine the number of gas phase reactants and products. The change in mols
+        # of gas will be used to determine Kc and Kp since solid species have reference
+        # concentration of 1
+        try:
+            gas_reacts = len([spcs for spcs in self.reactants if not spcs.contains_surface_site()])
+            gas_prods = len([spcs for spcs in self.products if not spcs.contains_surface_site()])
+        except IndexError:
+            logging.warning(f"Species do not have an rmgpy.molecule.Molecule "  
+                            "Cannot determine phases of species. We will assume "
+                            "ideal gas mixture when calculating Kc and Kp.")
+            gas_reacts = len(self.reactants)
+            gas_prods = len(self.products)
+
+        surf_reacts = len(self.reactants) - gas_reacts
+        surf_prods = len(self.products) - gas_prods
+        v_surf = surf_prods - surf_reacts # change in mols of surface spcs
```

Comment:
not a big fan of these variable names. I guess you were going for "nu" but the v makes me start asking "why vee? is it volume?". Maybe `dN_gas` or `delta_stoichiometry_gas` or `change_in_moles_of_gas`?